src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Utilities.psm1

```diff
@@ -0,0 +1,148 @@
+        if ($exceptionRaised)
+        {
+            Write-Verbose "Starting cleanup of created resources"
+            # cleaning up tenant resource groups with resources
+            $Global:CreatedResourceGroups | where {$_.Level -eq 1} | % {
+                try { Remove-ResourceGroup -ResourceGroupName $_.ResourceGroupName -SubscriptionId $_.SubscriptionId -Token $_.Token}
+                catch {}
+            }
+
+            # cleaning up subscriptions
+            for ( $i=$Global:CreatedSubscriptions.Count -1; $i -ge 0; $i--)
+            {
+                try { Remove-Subscription $Global:CreatedSubscriptions[$i].SubscriptionId}
+                catch {}
+            }
+
+            # cleaning up admin resource groups with subscriptions
+            $Global:CreatedResourceGroups | where {$_.Level -eq 0} | % {
+                try { Remove-ResourceGroup -ResourceGroupName $_.ResourceGroupName -SubscriptionId $_.SubscriptionId -Token $_.Token}
+                catch {}
+            }
+            Write-Verbose "Done with the cleanup of resources"
+        }
+
+        $elapsed = (Get-Date) - $startDateTime
+        Write-Verbose "Test Run Time MM:SS - $($elapsed.Minutes)m : $($elapsed.Seconds)s"
+
+        $Global:CreatedSubscriptions.Clear()
+        $Global:CreatedResourceGroups.Clear()
+    }
+}
+
+<#
+.SYNOPSIS
+Sets Certificate policy to trust all the certificates. This avoids passing -DisableCertValidataion for each cmdlet call when using self signed certificates
+#>
+function Ignore-SelfSignedCert
+{
+
+add-type @"
+    using System.Net;
+    using System.Security.Cryptography.X509Certificates;
+    public class TrustAllCertsPolicy : ICertificatePolicy {
+        public bool CheckValidationResult(
+            ServicePoint srvPoint, X509Certificate certificate,
+            WebRequest request, int certificateProblem) {
+            return true;
+        }
+    }
+"@
+    [System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertsPolicy
+
+    Write-Warning -Message "CertificatePolicy set to ignore all server certificate errors"
+}
+
+<#
+.SYNOPSIS
+Retires the specified job the given numer of times, waiting the given interval between tries
+
+.PARAMETER ScriptBlock
+The script block to execute. Must be a predicate (return true or false)
+
+.PARAMETER Argument
+Argument to pass to the script block
+
+.PARAMETER MaxTries
+The maximum number of times to retry
+
+.PARAMETER IntervalInSeconds
+The number of seconds to wait before retrying
+#>
+function Retry-Function
+{
+    param([ScriptBlock] $ScriptBlock, [Object] $Argument, [int] $MaxTries, [int] $IntervalInSeconds)
+
+    if ($IntervalInSeconds -eq 0) { $IntervalInSeconds = 60  }
+
+    $result = Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Argument;
+    $tries = 1;
+    while(( $result -ne $true) -and ($tries -le $MaxTries))
+    {
+        Start-Sleep -Seconds $IntervalInSeconds
```

Comment:
removing this Retry-Function from my utilities and using the Retry-Function from common.ps1